assets/js/upload.js

```diff
@@ -27,6 +27,27 @@ function elementForEmbeddedImage({ camo_url, type }) {
   return makeEl(tagName, { className: 'scraper-preview--image', src: objectUrl });
 }
 
+/**
+ * Execute the code when page was shown from back-forward cache.
+ * @param {() => void} callback
+ */
+function oncePersistedPageShown(callback) {
+  /**
+   * @param {PageTransitionEvent} event
+   */
+  function onPageShown(event) {
+    if (!event.persisted) {
+      return;
+    }
+
+    window.removeEventListener('pageshow', onPageShown);
+
+    callback();
+  }
+
+  window.addEventListener('pageshow', onPageShown);
+}
```

Comment:
Preferably add this function to a file in utils/ and add a test for it

Reply:
Moved implementation to the events.ts, added tests. Also replaced the add/removeEventListener pair with AbortSignal, because it's more elegant.